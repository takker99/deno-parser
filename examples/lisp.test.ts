import { Lisp } from "./lisp.ts";
import { assertSnapshot } from "@std/testing/snapshot";

Deno.test("lisp symbol", async (t) => {
  await assertSnapshot(t, Lisp.parse(`a`));
  await assertSnapshot(t, Lisp.parse(`kebab-case-symbol`));
  await assertSnapshot(t, Lisp.parse(`snake_case_symbol`));
  await assertSnapshot(t, Lisp.parse(`camelCaseSymbol`));
  await assertSnapshot(t, Lisp.parse(`a0`));
});

Deno.test("lisp number", async (t) => {
  await assertSnapshot(t, Lisp.parse(`1`));
  await assertSnapshot(t, Lisp.parse(`12`));
  await assertSnapshot(t, Lisp.parse(`345`));
  await assertSnapshot(t, Lisp.parse(`1234567890`));
});

Deno.test("lisp list", async (t) => {
  await assertSnapshot(t, Lisp.parse(`()`));
  await assertSnapshot(t, Lisp.parse(`(1)`));
  await assertSnapshot(t, Lisp.parse(`(1 2)`));
  await assertSnapshot(t, Lisp.parse(`(list 1 2)`));
  await assertSnapshot(t, Lisp.parse(`( list 1 2 )`));
  await assertSnapshot(
    t,
    Lisp.parse(
      `
(def inc (x)
  (add x 1))

(print (inc 3))
`,
    ),
  );
});
