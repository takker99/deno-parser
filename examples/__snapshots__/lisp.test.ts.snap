export const snapshot = {};

snapshot[`lisp symbol 1`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 2,
      index: 1,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 2,
          index: 1,
          line: 1,
        },
        name: "LispSymbol",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: "a",
      },
    ],
  },
}
`;

snapshot[`lisp symbol 2`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 18,
      index: 17,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 18,
          index: 17,
          line: 1,
        },
        name: "LispSymbol",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: "kebab-case-symbol",
      },
    ],
  },
}
`;

snapshot[`lisp symbol 3`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 18,
      index: 17,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 18,
          index: 17,
          line: 1,
        },
        name: "LispSymbol",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: "snake_case_symbol",
      },
    ],
  },
}
`;

snapshot[`lisp symbol 4`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 16,
      index: 15,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 16,
          index: 15,
          line: 1,
        },
        name: "LispSymbol",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: "camelCaseSymbol",
      },
    ],
  },
}
`;

snapshot[`lisp symbol 5`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 3,
      index: 2,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 3,
          index: 2,
          line: 1,
        },
        name: "LispSymbol",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: "a0",
      },
    ],
  },
}
`;

snapshot[`lisp number 1`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 2,
      index: 1,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 2,
          index: 1,
          line: 1,
        },
        name: "LispNumber",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: 1,
      },
    ],
  },
}
`;

snapshot[`lisp number 2`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 3,
      index: 2,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 3,
          index: 2,
          line: 1,
        },
        name: "LispNumber",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: 12,
      },
    ],
  },
}
`;

snapshot[`lisp number 3`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 4,
      index: 3,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 4,
          index: 3,
          line: 1,
        },
        name: "LispNumber",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: 345,
      },
    ],
  },
}
`;

snapshot[`lisp number 4`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 11,
      index: 10,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 11,
          index: 10,
          line: 1,
        },
        name: "LispNumber",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: 1234567890,
      },
    ],
  },
}
`;

snapshot[`lisp list 1`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 3,
      index: 2,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 3,
          index: 2,
          line: 1,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: [],
      },
    ],
  },
}
`;

snapshot[`lisp list 2`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 4,
      index: 3,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 4,
          index: 3,
          line: 1,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 3,
              index: 2,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 2,
              index: 1,
              line: 1,
            },
            type: "ParseNode",
            value: 1,
          },
        ],
      },
    ],
  },
}
`;

snapshot[`lisp list 3`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 6,
      index: 5,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 6,
          index: 5,
          line: 1,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 3,
              index: 2,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 2,
              index: 1,
              line: 1,
            },
            type: "ParseNode",
            value: 1,
          },
          {
            end: {
              column: 5,
              index: 4,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 4,
              index: 3,
              line: 1,
            },
            type: "ParseNode",
            value: 2,
          },
        ],
      },
    ],
  },
}
`;

snapshot[`lisp list 4`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 11,
      index: 10,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 11,
          index: 10,
          line: 1,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 6,
              index: 5,
              line: 1,
            },
            name: "LispSymbol",
            start: {
              column: 2,
              index: 1,
              line: 1,
            },
            type: "ParseNode",
            value: "list",
          },
          {
            end: {
              column: 8,
              index: 7,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 7,
              index: 6,
              line: 1,
            },
            type: "ParseNode",
            value: 1,
          },
          {
            end: {
              column: 10,
              index: 9,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 9,
              index: 8,
              line: 1,
            },
            type: "ParseNode",
            value: 2,
          },
        ],
      },
    ],
  },
}
`;

snapshot[`lisp list 5`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 13,
      index: 12,
      line: 1,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 13,
          index: 12,
          line: 1,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 0,
          line: 1,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 7,
              index: 6,
              line: 1,
            },
            name: "LispSymbol",
            start: {
              column: 3,
              index: 2,
              line: 1,
            },
            type: "ParseNode",
            value: "list",
          },
          {
            end: {
              column: 9,
              index: 8,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 8,
              index: 7,
              line: 1,
            },
            type: "ParseNode",
            value: 1,
          },
          {
            end: {
              column: 11,
              index: 10,
              line: 1,
            },
            name: "LispNumber",
            start: {
              column: 10,
              index: 9,
              line: 1,
            },
            type: "ParseNode",
            value: 2,
          },
        ],
      },
    ],
  },
}
`;

snapshot[`lisp list 6`] = `
{
  type: "ParseOK",
  value: {
    end: {
      column: 1,
      index: 44,
      line: 6,
    },
    name: "LispFile",
    start: {
      column: 1,
      index: 0,
      line: 1,
    },
    type: "ParseNode",
    value: [
      {
        end: {
          column: 13,
          index: 26,
          line: 3,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 1,
          line: 2,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 5,
              index: 5,
              line: 2,
            },
            name: "LispSymbol",
            start: {
              column: 2,
              index: 2,
              line: 2,
            },
            type: "ParseNode",
            value: "def",
          },
          {
            end: {
              column: 9,
              index: 9,
              line: 2,
            },
            name: "LispSymbol",
            start: {
              column: 6,
              index: 6,
              line: 2,
            },
            type: "ParseNode",
            value: "inc",
          },
          {
            end: {
              column: 13,
              index: 13,
              line: 2,
            },
            name: "LispList",
            start: {
              column: 10,
              index: 10,
              line: 2,
            },
            type: "ParseNode",
            value: [
              {
                end: {
                  column: 12,
                  index: 12,
                  line: 2,
                },
                name: "LispSymbol",
                start: {
                  column: 11,
                  index: 11,
                  line: 2,
                },
                type: "ParseNode",
                value: "x",
              },
            ],
          },
          {
            end: {
              column: 12,
              index: 25,
              line: 3,
            },
            name: "LispList",
            start: {
              column: 3,
              index: 16,
              line: 3,
            },
            type: "ParseNode",
            value: [
              {
                end: {
                  column: 7,
                  index: 20,
                  line: 3,
                },
                name: "LispSymbol",
                start: {
                  column: 4,
                  index: 17,
                  line: 3,
                },
                type: "ParseNode",
                value: "add",
              },
              {
                end: {
                  column: 9,
                  index: 22,
                  line: 3,
                },
                name: "LispSymbol",
                start: {
                  column: 8,
                  index: 21,
                  line: 3,
                },
                type: "ParseNode",
                value: "x",
              },
              {
                end: {
                  column: 11,
                  index: 24,
                  line: 3,
                },
                name: "LispNumber",
                start: {
                  column: 10,
                  index: 23,
                  line: 3,
                },
                type: "ParseNode",
                value: 1,
              },
            ],
          },
        ],
      },
      {
        end: {
          column: 16,
          index: 43,
          line: 5,
        },
        name: "LispList",
        start: {
          column: 1,
          index: 28,
          line: 5,
        },
        type: "ParseNode",
        value: [
          {
            end: {
              column: 7,
              index: 34,
              line: 5,
            },
            name: "LispSymbol",
            start: {
              column: 2,
              index: 29,
              line: 5,
            },
            type: "ParseNode",
            value: "print",
          },
          {
            end: {
              column: 15,
              index: 42,
              line: 5,
            },
            name: "LispList",
            start: {
              column: 8,
              index: 35,
              line: 5,
            },
            type: "ParseNode",
            value: [
              {
                end: {
                  column: 12,
                  index: 39,
                  line: 5,
                },
                name: "LispSymbol",
                start: {
                  column: 9,
                  index: 36,
                  line: 5,
                },
                type: "ParseNode",
                value: "inc",
              },
              {
                end: {
                  column: 14,
                  index: 41,
                  line: 5,
                },
                name: "LispNumber",
                start: {
                  column: 13,
                  index: 40,
                  line: 5,
                },
                type: "ParseNode",
                value: 3,
              },
            ],
          },
        ],
      },
    ],
  },
}
`;
